Enable more tests on CI

baronfel opened the pull request:
This PR enables a few tests what were blocked off, as well as converting the msbuild-4 tests to use toolsversion 15 and adding in explicit fsharp.core and system.runtime references as appropriate (ie. when the test wasn't explicitly testing a missing fsharp.core reference).

This is currently failing across most of the cloud providers, but I wanted to open it up as a base for discussion and ensure that I was thinking about things the correct way.

baronfel committed:
reenable integrations tests with msbuild projects

- reenable outputcache and invalidprojectfile testscases
- update the output.json for each of these cases was consistent with a) the changed stacktraces of the ProjectCrackerTool and b) the 'quitting...' info message
- fix regexes that were over-escaping the groups/characters at the end
- readd the linter test exceptions
- update errors test project file to version 15
- add system.runtime reference
- strip packagesdir paths

reenable tests:

- finddeclarations
- multiproj
- multiunsaved
- paramcompletion
- projectreload
- robustcommands
- symboluse
- test1json
- nofsharpcore
- uncompiledreferencedprojects

adjust baselines:
- for new information coming back from symboluse
- because there's a file parse notification now

update expected message from CI run

enricosada commented:
seems ok to me.

dunno why fsharp.core doesnt resolve from GAC in mono, but who care atm, is ok with `HintPath`

enricosada committed:
fix projects of tests

must specify the target framework, otherwise is v4.0 by default.

the project wasnt building either (`msbuild /t:Build`) with warning

```
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targe
ts(2106,5): warning MSB3274: The primary reference "FSharp.Core" could not be resolved because it was built against the
 ".NETFramework,Version=v4.5" framework. This is a higher version than the currently targeted framework ".NETFramework,
Version=v4.0". [e:\github\fsharp\FsAutoComplete\test\FsAutoComplete.IntegrationTests\ErrorTestsJson\Test1.fsproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targe
ts(2106,5): warning MSB3267: The primary reference "System.IO", which is a framework assembly, could not be resolved in
 the currently targeted framework. ".NETFramework,Version=v4.0". To resolve this problem, either remove the reference "
System.IO" or retarget your application to a framework version which contains "System.IO". [e:\github\fsharp\FsAutoComp
lete\test\FsAutoComplete.IntegrationTests\ErrorTestsJson\Test1.fsproj]
```

remove useless `System.Runtime` to minimize changes

enricosada committed:
install Microsoft Build Tools 2013 in appveyor

that's require to install MSBuild v12 assemblies in GAC, needed by FCS

enricosada commented:
@baronfel i think some errors are because MSbuild 12 is not installed in appveyor.
let me fix that

baronfel commented:
Looks like on appveyor we need to add fsharp.core and system.io references to a few old-style project files, and on linux we've got much larger problems: 

```
Failed to parse file: <absolute path removed>/FsAutoComplete.IntegrationTests/RobustCommands/Project/FileTwo.fs
Failed to parse file: /home/travis/build/fsharp/FsAutoComplete/test/FsAutoComplete.IntegrationTests/RobustCommands/Project/Program.fs
```
those aren't even in json blobs, those are coming from the background file parser if I'm reading this right

enricosada commented:
@baronfel i rebased and squashed the commits.

i fixed the projects, in win should be ok.

the `TargetFrameworkVersion` property was missing, and so it was using `net40` as default.
The `fsharp.core` was not loading because compiled with `net45`.
Running `msbuild /t:Build` of projects shown the warning, and the new dotnet-proj-info project parser was working ok (same results as msbuild), instead project cracker was trying to fix things

@baronfel i'll check ci for travis

baronfel commented:
Nice detective work!

enricosada committed:
print error message

enricosada commented:
appveyor is now green.

the travis error is strange, i'll print the expection message too.
if no clues, i'll disable the test on unix and merge it

enricosada committed:
on mono, a ThreadAbortException is raised if parsing is aborted

enricosada commented:
ok, on mono a `ThreadAbortException` is raised.

Dunno if that's because is not killed fast enough or another reason, but can be ignored i think

baronfel commented:
It looks like the base error message coming back from MSBuild is being normalized in different ways on windows and *nix.

baronfel commented:
Yeah it looks like on mono now we'll need to strip off the `in [MEMORY LOCATION]:[LINE NUMBER]` part on the mono newlines as part of the normalization now.

baronfel commented:
saved:
```
Microsoft.FSharp.Compiler.SourceCodeServices.ProjectCrackerTool.ProjectCrackerTool.FSharpProjectFileInfo..ctor(String fsprojFileName, FSharpOption`1 properties, FSharpOption`1 enableLogging)\n   at 
```

different:
```
Microsoft.FSharp.Compiler.SourceCodeServices.ProjectCrackerTool.ProjectCrackerTool+FSharpProjectFileInfo..ctor (System.String fsprojFileName, Microsoft.FSharp.Core.FSharpOption`1[T] properties, Microsoft.FSharp.Core.FSharpOption`1[T] enableLogging) [0x000ba] in <5b8fc032f73f2f25a745038332c08f5b>:0 \n  at 
```

differences I see: 
* type names are fully-qualified
* memory location and line number are present now

enricosada committed:
skip test on mono

enricosada commented:
yes. i'll disable for now on mono.
also the message is not exactly the same, so unless we rewrite the message completly, is not the same, but ihmo it's just a minor diff.

baronfel commented:
tracking the normalization issues in #339

enricosada commented:
naa, it doesnt matter that much. tests like that are few.

the text can be normalized or the message rewritten, and will be ok.

baronfel commented:
The test failure now is coming from the project type parsing, which is reading from a streamreader. Somehow the streamreader line is null? That's weird.

enricosada committed:
fix null ref on StreamReader.StreamReader

Krzysztof-Cieslak commented:
Green!

enricosada commented:
Fix it, one commit at time.

Awesome work @baronfel
